cmd/user-list/main.go

```diff
@@ -1,94 +1,36 @@
 package main
 
 import (
-	"fmt"
-	"strings"
-
 	"github.com/alecthomas/kong"
 
+	"github.com/edulinq/autograder/internal/api/users"
+	"github.com/edulinq/autograder/internal/cmd"
 	"github.com/edulinq/autograder/internal/config"
-	"github.com/edulinq/autograder/internal/db"
 	"github.com/edulinq/autograder/internal/log"
-	"github.com/edulinq/autograder/internal/model"
-	"github.com/edulinq/autograder/internal/util"
 )
 
 var args struct {
 	config.ConfigArgs
+	cmd.CommonOptions
 
-	Emails []string `help:"Optional list of users to limit the results to (leave empty for all users)." arg:"" optional:""`
-	Course string   `help:"Only include user enrolled in this course."`
-	Table  bool     `help:"Output data to stdout as a TSV." default:"false"`
+	Table bool `help:"Output data to stdout as a TSV." default:"false"`
 }
 
 func main() {
 	kong.Parse(&args,
-		kong.Description("List users on the server (default) or from the specified course."),
+		kong.Description("List users on the server."),
 	)
 
 	err := config.HandleConfigArgs(args.ConfigArgs)
 	if err != nil {
 		log.Fatal("Could not load config options.", err)
 	}
 
-	db.MustOpen()
-	defer db.MustClose()
-
-	if args.Course == "" {
-		listServerUsers(args.Emails, args.Table)
-	} else {
-		listCourseUsers(args.Emails, args.Course, args.Table)
-	}
-}
-
-func listServerUsers(emails []string, table bool) {
-	users, err := db.GetServerUsers()
-	if err != nil {
-		log.Fatal("Failed to get server users.", err)
-	}
-
-	if len(emails) > 0 {
-		newUsers := make(map[string]*model.ServerUser, len(emails))
-		for _, email := range emails {
-			newUsers[email] = users[email]
-		}
-
-		users = newUsers
-	}
-
-	if table {
-		fmt.Println(strings.Join(model.SERVER_USER_ROW_COLUMNS, "\t"))
-		for _, user := range users {
-			fmt.Println(strings.Join(user.MustToRow(), "\t"))
-		}
-	} else {
-		fmt.Println(util.MustToJSONIndent(users))
-	}
-}
-
-func listCourseUsers(emails []string, courseID string, table bool) {
-	course := db.MustGetCourse(courseID)
-
-	users, err := db.GetCourseUsers(course)
-	if err != nil {
-		log.Fatal("Failed to get course users.", err)
-	}
-
-	if len(emails) > 0 {
-		newUsers := make(map[string]*model.CourseUser, len(emails))
-		for _, email := range emails {
-			newUsers[email] = users[email]
-		}
-
-		users = newUsers
-	}
+	request := users.ListRequest{}
 
-	if table {
-		fmt.Println(strings.Join(model.COURSE_USER_ROW_COLUMNS, "\t"))
-		for _, user := range users {
-			fmt.Println(strings.Join(user.MustToRow(), "\t"))
-		}
+	if args.Table {
+		cmd.MustHandleCMDRequestAndExitFull(`users/list`, request, users.ListResponse{}, args.CommonOptions, cmd.ListServerUsersTable)
 	} else {
-		fmt.Println(util.MustToJSONIndent(users))
+		cmd.MustHandleCMDRequestAndExitFull(`users/list`, request, users.ListResponse{}, args.CommonOptions, nil)
 	}
```

Comment:
You generally don't want conditionals like this.
It's not the worst thing, but you generally want things to be as simple as you can.
Here, you are calling a decently complex function in each branch of the conditional,
and the code is mostly repeated (all except for the last param).
But, you could instead just have a simple conditional that changes the last parameter.

Like:
```
var printFunc any = nil;
if (args.Table) {
    printFunc = cmd.ListServerUsersTable;
}
```

Note that the conditional is simpler, has only one branch, and there is no repeated code.